.config/dependency-cruiser.js

```diff
@@ -0,0 +1,488 @@
+        //       criteria: { matchesFocus: true },
+        //       attributes: {
+        //         fillcolor: "lime",
+        //         penwidth: 2,
+        //       },
+        //     },
+        //     {
+        //       criteria: { matchesFocus: false },
+        //       attributes: {
+        //         fillcolor: "lightgrey",
+        //       },
+        //     },
+        //     {
+        //       criteria: { matchesReaches: true },
+        //       attributes: {
+        //         fillcolor: "lime",
+        //         penwidth: 2,
+        //       },
+        //     },
+        //     {
+        //       criteria: { matchesReaches: false },
+        //       attributes: {
+        //         fillcolor: "lightgrey",
+        //       },
+        //     },
+        //     {
+        //       criteria: { source: "^src/model" },
+        //       attributes: { fillcolor: "#ccccff" }
+        //     },
+        //     {
+        //       criteria: { source: "^src/view" },
+        //       attributes: { fillcolor: "#ccffcc" }
+        //     },
+        //   ],
+        //   dependencies: [
+        //     {
+        //       criteria: { "rules[0].severity": "error" },
+        //       attributes: { fontcolor: "red", color: "red" }
+        //     },
+        //     {
+        //       criteria: { "rules[0].severity": "warn" },
+        //       attributes: { fontcolor: "orange", color: "orange" }
+        //     },
+        //     {
+        //       criteria: { "rules[0].severity": "info" },
+        //       attributes: { fontcolor: "blue", color: "blue" }
+        //     },
+        //     {
+        //       criteria: { resolved: "^src/model" },
+        //       attributes: { color: "#0000ff77" }
+        //     },
+        //     {
+        //       criteria: { resolved: "^src/view" },
+        //       attributes: { color: "#00770077" }
+        //     }
+        //   ]
+        // }
+      },
+      archi: {
+        /* pattern of modules that can be consolidated in the high level
+          graphical dependency graph. If you use the high level graphical
+          dependency graph reporter (`archi`) you probably want to tweak
+          this collapsePattern to your situation.
+        */
+        collapsePattern:
+          '^(packages|src|lib|app|bin|test(s?)|spec(s?))/[^/]+|node_modules/(@[^/]+/[^/]+|[^/]+)',
+
+        /* Options to tweak the appearance of your graph.See
+           https://github.com/sverweij/dependency-cruiser/blob/main/doc/options-reference.md#reporteroptions
+           for details and some examples. If you don't specify a theme
+           for 'archi' dependency-cruiser will use the one specified in the
+           dot section (see above), if any, and otherwise use the default one.
+         */
+        // theme: {
+        // },
+      },
+      text: {
+        highlightFocused: true,
+      },
+    },
+  },
+}
+// generated: dependency-cruiser@15.5.0 on 2023-12-21T20:53:37.496Z
```

Comment:
@naugtur @legobeat 

Most of the config file was automatically generated by `dependency-cruiser`'s `init` command.  If helpful, I can show a diff of the generated config against my additions.

It _is_ a lot.  I get it.  Much of it is inscrutable.  It _is_ well-documented (in `dependency-cruiser`'s repo), however, and it seems to be able to handle whatever we can throw at it thus far.

And yeah: it will probably require some maintenance.  Think of it like a `policy.json`. 😜	

The tradeoff I am proposing is this: 

We absorb this complexity in order to enable **more robust guards against package and dependency misconfiguration _in a way tailored to the particulars and quirks of our project_**.

Reply:
The original problem (#793) this intends to solve is: we can publish broken packages, because our tooling is insufficient.

While I think it's important to have an understanding of the tooling changes, we should also weigh that against the problems we're looking to solve. Others, such as #836 or #669, are nice-to-have.  I wouldn't call this _critical_ by any means, but I don't have much confidence in the current state of our dependency checks.

I am open to using another solution that solves invalid dependencies, if it provides a similar level of coverage and is relatively easier to use & maintain.  We, of course, can rip `dependency-cruiser` out later if we don't like it.